[Leaderboards] Defined and added tiers to bar chart viz, added number rankings, changed viz effects of bar chart

thucd opened the pull request:
I defined and added tiers and corresponding badges for each metric (points, minutes played, and games won). I added badges and number rankings next to the names of each player. I also changed one of the visual effect of the bar chart by changing the bounce of the bar chart as it is loading. Addresses Issue #1354

Commit:
[leaderboards] updated the visualization of tiers and look of bar charts
- fixed the bounce of the bar charts so that they look less glitchy and more smooth
- added number rankings next to the player name
- defined tiers for each metric (points, minutes, games won) and corresponding badges
- added each player's badge next to their name
WIP: #1354

jgb4 commented on src/chigame/leaderboards/urls.py:
```diff
@@ -1,7 +1,11 @@
 urlpatterns = [
-    path("<int:game_id>/", leaderboard_view, name="leaderboard_view"),
+    path("<int:game_id>/", views.leaderboard_view, name="leaderboard_view"),
+    path("bar-chart/<int:game_id>/", views.bar_chart, name="bar_chart"),
```
I think this should be "bar-chart/points/<int:game_id>/" to match the style of the other two

thucd replied:
changed the url! thanks jeremy you can now view the bar charts for each game and still switch between tabs for each metric from these links:
http://127.0.0.1:8000/leaderboards/bar-chart/points/1/
http://127.0.0.1:8000/leaderboards/bar-chart/points/2/
http://127.0.0.1:8000/leaderboards/bar-chart/points/3/

Commit:
[leaderboards] addressed changes suggested by reviewer to change url name of points bar chart to be more descriptive and consistent with other bar chart url names PR ##1391, Issue #1354

jgb4 commented:
Thanks for resolving the urls comment. 

Two things to resolve:

1. For "Time Played" I think we definitely want to display the unit of measurement somewhere on the page.

2. There seems to be an issue where the bar charts are showing only two emojis (showing one of them twice), e.g.:
<img width="1511" alt="Screenshot 2025-05-23 at 8 35 43 AM" src="https://github.com/user-attachments/assets/f774fd84-0d23-4fe8-8c1d-fd1e8a08484d" />

Also, I'm having a problem with testing (I don't think it's unique to your PR). I am unable to see the leaderboards with the fixture data:
<img width="378" alt="Screenshot 2025-05-23 at 8 39 21 AM" src="https://github.com/user-attachments/assets/d8951573-20d5-44e4-a310-f777b1e27d91" />

Here is what I'm running in my terminal venv (from a clone of dev):
`git fetch origin pull/1391/head:pr-1391`
`git checkout pr-1391`
`python manage.py makemigrations`
`python manage.py makemigrations leaderboards`
`python manage.py migrate`
`run python3 manage.py loaddata test_leaderboard_fixture.json`
`run python3 manage.py runserver`

Any idea why I'm not seeing fixture data in the leaderboards?
Can I see what you are seeing at .../leaderboards/1/?

Thanks.

jgb4 reviewed (changes requested):
Pls address comments (1) and (2) thanks!

Commit:
[leaderboards] addressed changes suggested by reviewer to (1) add min unit of measurement and (2) change the scores that correspond to some tiers so that the demo shows each example player on a different tier

thucd commented:
thanks for the feedback jeremy! i addressed comments (1) and (2) and i created a separate issue #1430 to address the problem about seeing the leaderboards with the fixture data

jgb4 reviewed (approved):
Looks great, thanks for addressing my comments

thucd committed:
[leaderboards] fixed linter issues in bar_chart.html for PR #1391

thucd committed:
[leaderboards] fixed linter issues in views.py for PR #1391

thucd committed:
Merge branch 'dev' into leaderboards/bar-chart-tiers

thucd committed:
[leaderboards] fixing linter issues in views.py for PR #1391

thucd committed:
[leaderboards] fixing linter issues in bar_chart.html for PR #1391

thucd committed:
[leaderboards] fixing linter issues in bar_chart.html for PR #1391

thucd committed:
[leaderboards] fixing linter issues in views.py for PR #1391

thucd committed:
[leaderboards] fixing linter issues in urls.py for PR #1391

thucd committed:
[leaderboards] fixing linter issues in bar_chart.html for PR #1391

thucd committed:
[leaderboards] fixing linter issues in views.py for PR #1391

thucd committed:
[leaderboards] fixing linter issues in bar_chart.html for PR #1391

thucd committed:
[leaderboards] fixing linter issues in views.py for PR #1391

thucd committed:
[leaderboards] fixing linter issues in views.py for PR #1391

cmarshall282 committed:
Merge branch 'dev' into leaderboards/bar-chart-tiers

cmarshall282 committed:
[leaderboards] fixed template

cmarshall282 reviewed (approved):
Just like in #1379 I had to add some linter ignores. Eventually we want to add customizable tiers, but this is a great start!

cmarshall282 commented:
Issue Score: Excellent

Comments:
Great job! Here is some feedback:

- When you create a PR, you are requesting a review of the code in the branch you want to merge. As such, you should not push any commits to the branch right after creating the PR; you should only do so in response to a code review. Make sure that, when you create a PR, the code is ready for review and passes all the tests.
- 

<!-- {"score": "excellent", "minor": false} -->
